json_serializable/lib/src/enum_utils.dart

```diff
@@ -88,6 +88,20 @@ Object? _generateEntry({
 
       final e = fieldElementType.getField(valueField);
 
+      if (e == null && valueField == 'index') {
+        final values = fieldElementType.getField('values')!;
+
+        final items = ConstantReader(values.computeConstantValue()).listValue;
+
+        return items
+            .singleWhere((element) {
+              final name = element.getField('_name')!.toStringValue()!;
```

Comment:
is `_name` here something that is guaranteed to stay the same?

Reply:
I'm not sure what you are trying to achieve here. You have `FieldElement field` which is a constant in an `EnumElement`. Then you try to find the corresponding `DartObject` value in its fields, using the name, and get back its index in `values`. If I understand it correctly, then why not use `enumElement.fields.where((field) => field.isEnumConstant).toList().indexOf(fieldElement)`?

Reply:
Thanks! done!